bindings/flow/fdb_flow.actor.cpp

```diff
@@ -98,8 +98,83 @@ void fdb_flow_test() {
 }
 
 namespace FDB {
+class DatabaseImpl : public Database, NonCopyable {
+public:
+	virtual ~DatabaseImpl() { fdb_database_destroy(db); }
 
-	static inline void throw_on_error( fdb_error_t e ) {
+	Reference<Transaction> createTransaction() override;
+	void setDatabaseOption(FDBDatabaseOption option, Optional<StringRef> value = Optional<StringRef>()) override;
+
+private:
+	FDBDatabase* db;
+	explicit DatabaseImpl(FDBDatabase* db) : db(db) {}
+
+	friend class API;
+};
+
+class TransactionImpl : public Transaction, private NonCopyable, public FastAllocated<TransactionImpl> {
+	friend class DatabaseImpl;
+
+public:
+	virtual ~TransactionImpl() {
+		if (tr) {
+			fdb_transaction_destroy(tr);
+		}
+	}
+
+	void setReadVersion(Version v) override;
+	Future<Version> getReadVersion() override;
+
+	Future<Optional<FDBStandalone<ValueRef>>> get(const Key& key, bool snapshot = false) override;
+	Future<FDBStandalone<KeyRef>> getKey(const KeySelector& key, bool snapshot = false) override;
+
+	Future<Void> watch(const Key& key) override;
+
+	using Transaction::getRange;
+	Future<FDBStandalone<RangeResultRef>> getRange(const KeySelector& begin, const KeySelector& end,
+	                                               GetRangeLimits limits = GetRangeLimits(), bool snapshot = false,
+	                                               bool reverse = false,
+	                                               FDBStreamingMode streamingMode = FDB_STREAMING_MODE_SERIAL) override;
+
+	void addReadConflictRange(KeyRangeRef const& keys) override;
+	void addReadConflictKey(KeyRef const& key) override;
+	void addWriteConflictRange(KeyRangeRef const& keys) override;
+	void addWriteConflictKey(KeyRef const& key) override;
+
+	void atomicOp(const KeyRef& key, const ValueRef& operand, FDBMutationType operationType) override;
+	void set(const KeyRef& key, const ValueRef& value) override;
+	void clear(const KeyRangeRef& range) override;
+	void clear(const KeyRef& key) override;
+
+	Future<Void> commit() override;
+	Version getCommittedVersion() override;
+	Future<FDBStandalone<StringRef>> getVersionstamp() override;
+
+	void setOption(FDBTransactionOption option, Optional<StringRef> value = Optional<StringRef>()) override;
+
+	Future<Void> onError(Error const& e) override;
+
+	void cancel() override;
+	void reset() override;
+
+	TransactionImpl() : tr(NULL) {}
+	TransactionImpl(TransactionImpl&& r) noexcept(true) {
+		tr = r.tr;
+		r.tr = NULL;
+	}
+	TransactionImpl& operator=(TransactionImpl&& r) noexcept(true) {
+		tr = r.tr;
+		r.tr = NULL;
+		return *this;
+	}
+
+private:
+	FDBTransaction* tr;
+
+	explicit TransactionImpl(FDBDatabase* db);
+};
+
+    static inline void throw_on_error( fdb_error_t e ) {
```

Comment:
This is using spaces instead of tabs. Also, maybe not relevant since this isn't a public method, but the use of underscores in the name doesn't match any of the other methods. 

EDIT: I see from below that this name already existed, so you can probably ignore the name style if you want.